### PR TITLE
[cvss3] Use strongly typed enums for cvss3 vector

### PR DESCRIPTION
**Pull request by n1chre**
Improved version of ./cvss/v3
Test: go test

### CODE REVIEW

File: cvss3/base_metrics.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+func (av AttackVector) Weight() float64 {
+	return weightsAttackVector[av]
+}
+
+func (av AttackVector) String() string {
+	return stringAttackVector[av]
+}
+
+func (av *AttackVector) Parse(str string) error {
```
Comment (skogtwin) [inline cvss3/base_metrics.go]: Same, seems to be only used internally.

File: cvss3/base_metrics.go [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+	stringAttackVector  = []string{"", "N", "A", "L", "P"}
+	mapAttackVector     = map[string]AttackVector{
+		"N": AttackVectorNetwork,
+		"A": AttackVectorAdjecent,
+		"L": AttackVectorLocal,
+		"P": AttackVectorPhysical,
+	}
+)
+
+func (av AttackVector) Defined() bool {
```
Comment (skogtwin) [inline cvss3/base_metrics.go]: Is it supposed to be used outside `package cvss3`?

File: cvss3/score.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import (
+	"math"
+)
+
+func roundUp(x float64) float64 {
+	// round up to one decimal
+	return math.Ceil(x*10) / 10
+}
+
+// // Score = combined score for the whole Vector
```
Comment (skogtwin) [inline cvss3/score.go]: delet :)

File: cvss3/vector.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	first := true
+	for _, metric := range order {
+		def := defineables[metric]
+		if !def.Defined() {
+			continue
+		}
+		if !first {
+			fmt.Fprintf(sb, partSeparator)
```
Comment (skogtwin) [inline cvss3/vector.go]: you probably meant `fmt.Fprint`

File: cvss3/vector.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+		} else {
+			first = false
+		}
+		fmt.Fprintf(sb, "%s%s%s", metric, metricSeparator, def)
+	}
+
+	return prefix + sb.String()
```
Comment (skogtwin) [inline cvss3/vector.go]: Why not write `prefix` into `sb` at the start of this function instead?

File: cvss3/base_metrics.go [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+	return int(av) != 0
+}
+
+func (av AttackVector) Weight() float64 {
```
[the remark this answers is not in the document]
Comment (n1chre) [reply cvss3/base_metrics.go]: I'll leave the weight function, but make it lowercase

File: cvss3/base_metrics.go [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+	return weightsAttackVector[av]
+}
+
+func (av AttackVector) String() string {
```
[the remark this answers is not in the document]
Comment (n1chre) [reply cvss3/base_metrics.go]: Using it in output

File: cvss3/base_metrics.go (hunk 2 above)
Comment (n1chre) [reply cvss3/base_metrics.go]: no

**Pull request merged**